crates/matrix-sdk-base/src/store/traits.rs

```diff
@@ -634,23 +645,34 @@ pub enum StateStoreDataValue {
 
     /// A filter with the given ID.
     Filter(String),
+
+    /// The user avatar url
+    UserAvatarURL(String),
 }
 
 impl StateStoreDataValue {
     /// Get this value if it is a sync token.
     pub fn into_sync_token(self) -> Option<String> {
-        match self {
-            Self::SyncToken(token) => Some(token),
-            Self::Filter(_) => None,
-        }
+        let Self::SyncToken(token) = self else {
+            return None;
+        };
+        Some(token)
```

Comment:
Why did you rewrite this method?

Reply:
every time a new type is added we needed to add a new match arm, also only one single match arm is required for these functions.

Reply:
Right, for that you can just change the second branch from `Self::Filter(_)` to `_` though.

Reply:
yeah I just preferred the` let else` syntax, but I guess is a personal thing

Reply:
done!